tests/apps/perf/configuration/app.go

```diff
@@ -0,0 +1,523 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	"github.com/redis/go-redis/v9"
+
+	runtimev1pb "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/dapr/tests/apps/utils"
+)
+
+const (
+	daprHost        = "localhost"
+	daprHTTPPort    = "3500"
+	daprGRPCPort    = 50001
+	configStoreName = "configstore"
+	separator       = "||"
+	redisHost       = "dapr-redis-master.dapr-tests.svc.cluster.local:6379"
+	writeTimeout    = 5 * time.Second
+)
+
+var (
+	appPort              = 3000
+	updater              Updater
+	httpClient           = utils.NewHTTPClient()
+	grpcClient           runtimev1pb.DaprClient
+	notifyCh             = make(chan struct{})
+	subscribeStopChanMap = make(map[string]map[string]chan struct{})
+)
+
+type appResponse struct {
+	Message   string `json:"message,omitempty"`
+	StartTime int    `json:"start_time,omitempty"`
+	EndTime   int    `json:"end_time,omitempty"`
+}
+
+type receivedMessagesResponse struct {
+	ReceivedUpdates []string `json:"received-messages"`
+}
+
+type Item struct {
+	Value    string            `json:"value,omitempty"`
+	Version  string            `json:"version,omitempty"`
+	Metadata map[string]string `json:"metadata,omitempty"`
+}
+
+type UpdateEvent struct {
+	ID    string           `json:"id"`
+	Items map[string]*Item `json:"items"`
+}
+
+type Updater interface {
+	Init() error
+	Update(items map[string]*Item) error
+	Get(key string) (string, error)
+	Subscribe(keys []string) (string, error)
+}
+
+type RedisUpdater struct {
+	client *redis.Client
+}
+
+func init() {
+	p := os.Getenv("PORT")
+	if p != "" && p != "0" {
+		appPort, _ = strconv.Atoi(p)
+	}
+} // indexHandler is the handler for root path
+func indexHandler(w http.ResponseWriter, r *http.Request) {
+	log.Println("indexHandler is called")
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(appResponse{Message: "OK"})
+}
+
+// sendResponse returns response with status code and message
+func sendResponse(w http.ResponseWriter, statusCode int, message string) {
+	w.WriteHeader(statusCode)
+	json.NewEncoder(w).Encode(appResponse{Message: message})
+}
+
+// return key-value pairs as a list of strings
+func getRedisValuesFromItems(items map[string]*Item) []interface{} {
+	m := make([]interface{}, 0, 2*len(items)+1)
+	for key, item := range items {
+		val := item.Value + separator + item.Version
+		m = append(m, key, val)
```

Comment:
nitpick: Not needed in test code, but avoid append and use indexing where possible.